ptyprocess/ptyprocess.py

```diff
@@ -206,6 +206,19 @@ def spawn(
         if not isinstance(argv, (list, tuple)):
             raise TypeError("Expected a list or tuple for argv, got %r" % argv)
 
+        # env vars must be utf-8 encoded strings
+        # https://github.com/pexpect/pexpect/issues/512
+        #
+        # from https://docs.python.org/3.8/library/os.html#os.execvpe
+        # Errors will be reported as OSError exceptions.
+        if isinstance(env, dict):
+            _text_type = str
+            if not PY3:
+                _text_type = unicode
+            for k, v in env.items():
+                if isinstance(v, _text_type):
+                    env[k] = v.encode('utf-8')
```

Comment:
IIRC modifying a dict while iterating over it is generally considered a bad idea. Maybe it's better to create a new dict `env_b` instead.